tests: CLI

a-mpch opened the pull request:
Given  #193  introduced a bug given a data directory and was fixed in #198 

Added some unit test to reading certificates so we can check in the CI this kind of errors

dunxen commented on src/cli.rs:
```diff
@@ -345,36 +345,47 @@ fn read_macaroon_from_file(path: PathBuf) -> Result<String, std::io::Error> {
+    match read_cert_from_args(cert_pem, cert_path) {
+        Ok(config) => config,
+        Err(err) => {
+            println!("{}", err);
+            exit(1);
```
pre-existing, but i believe we can do something more idiomatic when exiting and tap into clap's errors for output. not needed in this PR.

a-mpch replied:
added an issue for this https://github.com/lndk-org/lndk/issues/202

dunxen reviewed (commented):
Thanks for this! It LGTM. Just checking if @orbitalturtle had any thoughts on what else might need covering.

a-mpch commented on src/cli.rs:
```diff
@@ -407,3 +418,108 @@ fn read_macaroon_from_args(
+                None => std::env::remove_var(self.key),
+            }
+        }
+    }
```
thanks!

orbitalturtle commented:
@dunxen Thanks! I'll give this a look tomorrow

orbitalturtle commented on src/cli.rs:
```diff
@@ -407,3 +418,108 @@ fn read_macaroon_from_args(
+        };
+
+        assert!(result.is_ok());
+    }
```
nit: missing line break here

orbitalturtle reviewed (commented):
Two super-micro nits, but otherwise looks great 🎊, the EnvironmentGuard is super nifty.

Would you mind adding a similar set of unit tests for `read_macaroon_from_args` just so we have that covered as well?

a-mpch commented:
thanks for both reviews @dunxen @orbitalturtle! I rebased on top of master and corrected nit comments.

>EDIT: Oh I just realized, these are tests are really great to have, but I don't think they quite prevent the scenario in https://github.com/lndk-org/lndk/pull/198 because in https://github.com/lndk-org/lndk/pull/198 the problem was that when the lndk binary was run, it wasn't putting the certs and logs in the correct location, and this doesn't quite test that I don't think
>This could be a future PR, but I think we'd want to add some sort of a quick integration test that spins up LNDK + simply checks that lndk.log, tls-cert.pem, and  tls-key.pem are located in ~/.lndk, where we expect. I'll create an issue for this soon.

yes! totally. I'm fine doing that after the ones I'm currently working on. re-reading the PR 198, we were creating TLS on the correct place and the CLI wasn't updated to get that. In a scenario of this tests would be running in the first changes, it would alert those changes because is kinda of hard coded where should go.
Anyway, a itest checking the scenario would be 100% better and feel free to assign it to me when writing the issue!

orbitalturtle commented:
Sounds good, thanks @a-mpch!!
